terraform/modules/quorum-cluster-region/bootnodes.tf

```diff
@@ -265,15 +265,15 @@ EOF
 # BOOTNODE IAM POLICY ATTACHMENT AND INSTANCE PROFILE
 # ---------------------------------------------------------------------------------------------------------------------
 resource "aws_iam_role_policy_attachment" "bootnode" {
-  count = "${signum(lookup(var.bootnode_counts, var.aws_region, 0))}"
+  count = "${lookup(var.bootnode_counts, var.aws_region, 0)}"
 
-  role       = "${aws_iam_role.bootnode.name}"
+  role       = "${element(aws_iam_role.bootnode.*.name, count.index)}"
   policy_arn = "${aws_iam_policy.quorum.arn}"
 }
 
 resource "aws_iam_instance_profile" "bootnode" {
-  count = "${signum(lookup(var.bootnode_counts, var.aws_region, 0))}"
+  count = "${lookup(var.bootnode_counts, var.aws_region, 0)}"
 
-  name = "bootnode-${var.aws_region}-network-${var.network_id}"
-  role = "${aws_iam_role.bootnode.name}"
+  name = "bootnode-${var.aws_region}-network-${var.network_id}-bootnode-${count.index}"
```

Comment:
Looks like you changed the name of the role but didn't change this to match. Preferably they'd be the same.

You could also interpolate it (which may be what I should have done in the first place) like `name = ${element(aws_iam_role.bootnode.*.name, count.index)}`